src/array.ts

```diff
@@ -219,6 +221,29 @@ Array.prototype.asyncFilter = async function <T>(
   return this.filter((_v, index) => results[index]);
 };
 
+Array.prototype.asyncSome = async function <T>(
+  this: Array<T>,
+  predicate: (v: T) => boolean | Promise<boolean>,
+): Promise<boolean> {
+  const asyncResults: Promise<boolean>[] = [];
+  for (const el of this) {
+    const result = predicate(el);
+    if (typeof result === "boolean") {
+      if (result) {
+        return true;
+      }
+    } else {
+      asyncResults.push(result);
+    }
+  }
+  for (const result of asyncResults) {
+    if (await result) {
+      return true;
+    }
+  }
+  return false;
+};
```

Comment:
I had a feeling that there was a much easier way of doing that boolean type guard :joy:, making the change now!

Reply:
Just pushed the changes. I was able to mostly use your suggestion, just had to add an `instanceof` check before pushing to `asyncResults` because it could potentially be `false`.